src/argilla/client/sdk/datasets/models.py

```diff
@@ -36,7 +36,7 @@ def _missing_(cls, value):
 class BaseDatasetModel(BaseModel):
     tags: Dict[str, str] = Field(default_factory=dict)
     metadata: Dict[str, Any] = Field(default_factory=dict)
-    name: str = Field(regex="^(?!-|_)[a-z0-9-_]+$")
+    name: str = Field(regex="^(?!-|_)[a-zA-Z0-9-_]+$")
```

Comment:
I also updated this regex, but didn't include the space just in case

Reply:
If that's the old Dataset model I think we should not change it. Correct me if I'm wrong @frascuchon

Reply:
Makes sense, but was that intended originally? I mean, not allowing upper-case letters in the dataset name

Reply:
The idea was to format the dataset name so it can be used as ID for ElasticSearch indexes.